pkgs/development/misc/google-clasp/default.nix

```diff
@@ -0,0 +1,19 @@
+{ stdenv, pkgs }:
+let
+  version = "2.2.1";
+in
+(import ./google-clasp.nix {
+  inherit pkgs;
+  inherit (stdenv.hostPlatform) system;
+})."@google/clasp-${version}".override rec {
+  preRebuild = ''
+    patch -p1 <<<"${builtins.readFile ./dotf.patch}"
+  '';
```

Comment:
Thanks @turboMaCk. The unpacking of node packages takes place in the `installPhase`:

```
checking outputs of '/nix/store/bd9fs5z96ypszcn0zqmh9lp2l29m51sp-node-_at_google_slash_clasp-2.2.1.drv'...                                                                  
unpacking sources                                                                                                                                                            
patching sources                                                                                                                                                             
configuring                                                                                                                                                                 
no configure script, doing nothing                                                                                                                                           
building                                                                                                                                                                     
installing                                                                                                                                                                   
unpacking source archive /nix/store/xzbw3n2j4zq5nh1j4dkx0aqn1mr629lp-clasp-2.2.1.tgz                                                                                         
unpacking source archive /nix/store/ibwdid42ly8x7xidcvp0jr7jnnm281lf-fs.scandir-2.1.1.tgz                                                                                    
unpacking source archive /nix/store/zfpb0hszl9827qmhbkhy3sksrlyfnja0-fs.stat-2.0.1.tgz                                                                                       
unpacking source archive /nix/store/33qz58lq0f86hcbdi13jp04ndwp4c824-fs.walk-1.2.2.tgz                                                                               
...
```

The `patchPhase` does not have the sources available yet.

The nodePackages seem to patch the sources during in the `preBuild` phase:

- https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/default-v10.nix#L47
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/default-v10.nix#L71

So this can be inconsistent either with a regular nix package build or node-package build. An alternative approach to this would be:
1. list the package in [node-packages-v10.json ](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/node-packages-v10.json)
2. add another override like ([pnpm = nodePackages.pnpm.override](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/default-v10.nix#L67)) in the [default-v10.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/default-v10.nix)

I don't have a strong opinion on this one.

@adisbladis toughts?

Reply:
I was playing with this and I think the solution you come up with is fine and most straight forward at the moment.

Reply:
Seems fine for now. However I'd like to refresh my knowledge about the node2nix codebase first to check if there's a proper workaround (otherwise I'd file an issue there later).